src/AutoRest.CSharp/Common/Generation/Writers/RequestWriterHelpers.cs

```diff
@@ -334,21 +334,39 @@ private static void WriteSerializationFormat(CodeWriter writer, SerializationFor
         private static void WriteQueryParameter(CodeWriter writer, CodeWriterDeclaration uri, QueryParameter queryParameter)
         {
             string? delimiter = GetSerializationStyleDelimiter(queryParameter.SerializationStyle);
-            string method = delimiter != null
+            bool explode = queryParameter.Explode;
+            string method = delimiter != null && !explode
                 ? nameof(RequestUriBuilderExtensions.AppendQueryDelimited)
                 : nameof(RequestUriBuilderExtensions.AppendQuery);
 
             ReferenceOrConstant value = queryParameter.Value;
             using (WriteValueNullCheck(writer, value))
             {
-                writer.Append($"{uri}.{method}({queryParameter.Name:L}, ");
-                WriteConstantOrParameter(writer, value, enumAsString: true);
-                if (delimiter != null)
+                if (explode)
                 {
-                    writer.Append($", {delimiter:L}");
+                    var paramVariable = new CodeWriterDeclaration("param");
+                    writer.Append($"foreach(var {paramVariable:D} in ");
+                    WriteConstantOrParameter(writer, value, enumAsString: true);
+                    writer.Line($")");
+                    using (writer.Scope())
+                    {
+                        writer.Append($"{uri}.{method}({queryParameter.Name:L}, ");
+                        WriteConstantOrParameter(writer, new Reference("param", value.Type.Arguments.Length > 0 ? value.Type.Arguments[0]: value.Type), enumAsString: true);
```

Comment:
We don't need a reference here cause we know exactly what we are iterating. `writer.Append($"{uri}.{method}({queryParameter.Name:L}, {paramVariable}");` should be enough

Reply:
@AlexanderSher: that's not enough because .toString() won't be generated for enum types when use it directly as paramVariable.